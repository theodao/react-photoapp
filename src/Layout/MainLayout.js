import React, { useState, useEffect } from 'react';
import styled from 'styled-components';
import { Loader, TopMenu } from '@gotitinc/design-system';
import { withRouter } from 'react-router-dom';
import Button from '../components/Button';

const MainLayout = ({ loading, children, history, match }) => {
  const [current, setCurrent] = useState(match.path || '/dashboard');

  return (
    <div>
      {loading ? (
        <Container>
          <Loader />
        </Container>
      ) : (
        <>
          <div
            style={{
              display: 'flex',
              justifyContent: 'space-between',
            }}
          >
            <div
              style={{
                marginLeft: '10px',
              }}
            >
              <TopMenu current={current} onSelect={() => {}}>
                <TopMenu.Item
                  eventKey="/dashboard"
                  onClick={() => history.push('/dashboard')}
                >
                  Dashboard
                </TopMenu.Item>
                <TopMenu.Item
                  eventKey="/add-category"
                  onClick={() => history.push('/add-category')}
                >
                  Add categories
                </TopMenu.Item>
                <TopMenu.Item
                  eventKey="/add-item"
                  onClick={() => history.push('/add-item')}
                >
                  Add Items
                </TopMenu.Item>
              </TopMenu>
            </div>
            <div>
              <Button
                label="Log out"
                style={{
                  margin: '10px',
                }}
              />
            </div>
          </div>
          <div>{children}</div>
        </>
      )}
    </div>
  );
};

export default withRouter(MainLayout);

const Container = styled.div`
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100vh;
  background: rgba(0, 0, 0, 0.8);
  z-index: 999;
  display: flex;
  justify-content: center;
  align-items: center;
`;
