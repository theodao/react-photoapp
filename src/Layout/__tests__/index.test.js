import React from 'react';
// import { cleanup, render } from '@testing-library/react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { MainLayout } from '../MainLayout';

describe('Testing Mainlayout component', () => {
  // afterEach(cleanup);

  const pushFunction = jest.fn();

  const history = {
    push: pushFunction,
  };

  it('Should match snapshot', () => {
    const pushFunction = jest.fn();

    const history = {
      push: pushFunction,
    };

    const tree = renderer.create(<MainLayout match={{ path: '' }} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Should push to dashboard when clicking dashboard', () => {
    const wrapper = shallow(
      <MainLayout match={{ path: '' }} history={history} />,
    );

    wrapper
      .findWhere(
        (c) =>
          c.name() === 'TopMenu.Item' && c.prop('eventKey') === '/dashboard',
      )
      .props()
      .onClick();
    expect(pushFunction).toHaveBeenCalled();
  });

  it('Should push to add category when clicking add category', () => {
    const wrapper = shallow(
      <MainLayout match={{ path: '' }} history={history} />,
    );

    wrapper
      .findWhere(
        (c) =>
          c.name() === 'TopMenu.Item' && c.prop('eventKey') === '/add-category',
      )
      .props()
      .onClick();
    expect(pushFunction).toHaveBeenCalled();
  });

  it('Should push to add item when clicking to add item', () => {
    const wrapper = shallow(
      <MainLayout match={{ path: '' }} history={history} />,
    );

    wrapper
      .findWhere(
        (c) =>
          c.name() === 'TopMenu.Item' && c.prop('eventKey') === '/add-item',
      )
      .props()
      .onClick();
    expect(pushFunction).toHaveBeenCalled();
  });
});
