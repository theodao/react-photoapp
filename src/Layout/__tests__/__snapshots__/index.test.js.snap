// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Mainlayout component Should match snapshot 1`] = `
<div>
  <div
    style={
      Object {
        "display": "flex",
        "justifyContent": "space-between",
      }
    }
  >
    <div
      style={
        Object {
          "marginLeft": "10px",
        }
      }
    >
      <div
        className="TopMenu u-backgroundWhite u-paddingVerticalExtraSmall"
      >
        <div
          className="TopMenu-list"
        >
          <div
            className="TopMenu-item u-positionRelative is-active u-flexInline u-alignItemsCenter u-cursorPointer"
            onClick={[Function]}
          >
            <div
              className="TopMenu-itemBefore u-positionAbsolute u-positionLeft u-positionTop u-backgroundTransparent u-widthFull"
            />
            <div
              className="TopMenu-itemAfter u-positionAbsolute u-positionLeft u-positionBottom u-backgroundPrimary u-widthFull"
            />
            <span
              className="u-positionRelative u-flexInline u-flexGrow-1 u-paddingVerticalTiny hover:u-textDecorationNone u-textPrimary"
              onClick={[Function]}
            >
              <div
                className="u-flexGrow-1"
              >
                Dashboard
              </div>
            </span>
          </div>
          <div
            className="TopMenu-item u-positionRelative u-marginLeftLarge u-flexInline u-alignItemsCenter u-cursorPointer"
            onClick={[Function]}
          >
            <a
              className="u-positionRelative u-flexInline u-flexGrow-1 u-paddingVerticalTiny hover:u-textDecorationNone u-textDark hover:u-textPrimary"
              href="#"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="button"
            >
              <div
                className="u-flexGrow-1"
              >
                Add categories
              </div>
            </a>
          </div>
          <div
            className="TopMenu-item u-positionRelative u-marginLeftLarge u-flexInline u-alignItemsCenter u-cursorPointer"
            onClick={[Function]}
          >
            <a
              className="u-positionRelative u-flexInline u-flexGrow-1 u-paddingVerticalTiny hover:u-textDecorationNone u-textDark hover:u-textPrimary"
              href="#"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="button"
            >
              <div
                className="u-flexGrow-1"
              >
                Add Items
              </div>
            </a>
          </div>
        </div>
      </div>
    </div>
    <div>
      <button
        className="Button u-flexInline u-justifyContentCenter u-alignItemsCenter u-textDecorationNone u-roundedMedium u-textWhite hover:u-textWhite u-backgroundPrimary hover:u-backgroundPrimaryDark u-border u-borderPrimary hover:u-textDecorationNone u-cursorPointer u-textUppercase"
        disabled={false}
        onClick={[Function]}
        style={
          Object {
            "margin": "10px",
          }
        }
      >
        <span
          className="Button-label u-inlineBlock"
        >
          Log out
        </span>
      </button>
    </div>
  </div>
  <div />
</div>
`;
